runtime/near-vm-logic/src/logic.rs

```diff
@@ -2480,8 +2553,24 @@ impl<'a> VMLogic<'a> {
         }))
     }
 
-    /// Computes the outcome of execution.
-    pub fn outcome(self) -> VMOutcome {
+    /// Computes the outcome of the execution.
+    ///
+    /// If `FunctionCallWeight` protocol feature (127) is enabled, unused gas will be
+    /// distributed to functions that specify a gas weight. If there are no functions with
+    /// a gas weight, the outcome will contain unused gas as usual.
+    #[allow(unused_mut)]
```

Comment:
This function is used without the feature, are you suggesting duplicating the function logic and calling the respective functions based on if the feature is set or not? Seems a bit jankier this way

Reply:
Ah my bad. I meant 

```rust
#[cfg_attr(not(feature = "protocol_feature_function_call_weight"), allow(unused_mut))]
```

`cfg_attr` here applies the `allow(unused_mut)` based on absence of the `protocol_feature_function_call_weight` feature. Since the `mut` is only unused when `cfg!(feature = "protocol_feature_function_call_weight")`. This has the benefit of making it less easy to forget to remove the lint disable later when we remove the feature altogether and also allows to check that the rest of the code in the function does not accidentally violate this lint in future edits.

Reply:
oh oops, yes, I misread when skimming as well. Of course